### ISSUE TITLE
Notes about Windows installation

### ISSUE THREAD

**Issue by sebbacon** [ebmdatalab/custom-docker#4]
# Which software

There are two products: [Docker Desktop for Windows](https://docs.docker.com/docker-for-windows/install/) and [Docker Toolbox for Windows](https://docs.docker.com/toolbox/toolbox_install_windows/).

Docker Desktop offers native support via Hyper-V containers, and so is preferred, but requires Windows 10 64-bit Pro, Enterprise, or Education (Build 15063 or later), and Hyper-V and Containers Windows features must be enabled (all of which are the case on our standard university laptop installs).

Docker Toolbox runs docker within a Linux virtualbox via [Docker Machine](https://docs.docker.com/machine/overview/), and therefore offers a functional but sub-optimal experience.

You will find online different people complaining about support for both of these.

## Docker Desktop

I did an install and followed the default settings. On logging in again I was asked to enable Hyper-V and Containers, which I did.  This required another restart.

When starting Docker, it takes a while to actually start up - up to 5 minutes. While it's doing so, an animation runs in the notification area:

![image](https://user-images.githubusercontent.com/211271/72052991-14a8c000-32be-11ea-948f-575a3c84bc3b.png)

Another notification appears when it's finished.

"Running" means there's a docker service running on your computer, to which you can connect using the command line. You can check it's up and running by opening a Command Prompt and entering `docker info`, which should output a load of diagnostics.

To be able to access the windows filesystem from the docker container (and therefore do development inside Jupyter with results appearing in a place visible to Git), you must explicitly share your hard drive in the Docker settings (click system tray docker icon; select "settings"; select "shared drives")

## Network login issues

When installing from the office, and logged in as a network user, there have been permission problems that have been solved by adding the special "Authenticated Users" group to the `docker-users` group, per [this comment](https://github.com/docker/for-win/issues/785#issuecomment-327237998) (screenshot of place to do it [here](https://github.com/docker/for-win/issues/785#issuecomment-344805180)).

I've been unable to reproduce this at home because I've found it basically impossible to get the setup working when connected to the office VPN using Cisco Anyconnect (more below).

I have been able to make everything work when  logged in as a local user (i.e. to the computer only).

By way of notes on progress, it appears that something is blocking communication between the vEthernet addresses used by docker on IP addresses 10.0.75.1 and 10.0.75.2; however, this is *not* related to Windows Defender Firewall, as the problem continues to manifest even with the firewall completely switched off.  There are unsolved reports along the same lines online, e.g. [this one](https://forums.docker.com/t/issues-with-cisco-anyconnect-vpn/16206/6). Currently the only fix is to run without any connection to the VPN, although [following on from suggestions here](https://www.lullabot.com/articles/fixing-docker-and-vpn-ip-address-conflicts), I have tried  changing "Internal Virtual Switch" in the Docker Settings > Network to `172.30.0.0` without joy.

Finally, note that when authentication changes (e.g. different logins), you sometimes have to reauthorise Docker's "Shared Drives"  (click system tray docker icon; select "settings"; select "shared drives"; click Reset credentials; retick the drive to share; Apply)

**Comment by NickCEBM** [comment]
Two notes about windows set-up:

1. When your computer asks for permission to access the C drive, when you click to allow this, you should get a log-in window to appear. This doesn't always come to the forefront but will show up in your toolbar with the Docker icon. Make sure you enter this.

2. You will know the session started successfully when you see something like the following tags on the command line output:

![image](https://user-images.githubusercontent.com/25153969/72539653-8ca35700-3877-11ea-85d4-214acb25f7e2.png)

**Comment by brianmackenna** [comment]
Two further notes from my installation (that went relatively smoothly) to support updating readme 
- *Hyper-V*  - when I installed docker I got a message to say hyper-V is not enabled. This is well described from a quick google and I used this [youtube video](https://www.youtube.com/watch?v=m6DX9B-xtVU) to find the Hyper-V and turn it on. Weirdly my HyperV was enabled but ticking and unticking the box, followed by a restart appears to have fixed the problem. 
- access permissions : I'll admit I got a little click happy during installation and wondering did I set the security access correctly based on warnings below. I think I can see  them in @NickCEBM screenshot above as well!

![image](https://user-images.githubusercontent.com/40460354/74263540-f38d2200-4cf6-11ea-9ba6-1cfd8d4ead81.png)
